### PR TITLE
New Cellranger-atac modules

### PR DESCRIPTION
**Pull request by heylf**
## PR checklist

Closes #XXX 

- [X] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [X] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [X] Remove all TODO statements.
- [X] Emit the `versions.yml` file.
- [X] Follow the naming conventions.
- [X] Follow the parameters requirements.
- [X] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by heylf** [comment]
Waiting for docker containers by nf-core.

**Comment by maxulysse** [comment]
@heylf 
There are a couple of merge conflicts, I'm guessing mainly due to out of date code.
sorry for not seeing this PR earlier.

Can you try to solve the conflicts?

**Comment by heylf** [comment]
@maxulysse the conflict rely with missing containers mainly. I have some containers ready but they need to be made available by nf-core, because of policies. I will push for this again.

**Comment by maxulysse** [comment]
Let me know if I can help in any way

File: modules/nf-core/cellrangeratac/mkfastq/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    && apt-get install -y \
+    cpio \
+    procps \
+    rpm2cpio \
+    unzip \
+    && apt-get clean -y && rm -rf /var/lib/apt/lists/*
+
+# Copy pre-downloaded bcl2fastq2 and cellranger-atac file
+ENV BCL2FASTQ2_VER=v2-20-0-linux-x86-64 \
+    ATAC_VER=2.1.0
```
Comment (maxulysse) [inline modules/nf-core/cellrangeratac/mkfastq/Dockerfile:20]:
```suggestion
    CELLRANGERATAC_VER=2.1.0
```

File: modules/nf-core/cellrangeratac/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Dockerfile to create container with Cell Ranger ATAC v2.1.0
+# Push to nfcore/cellrangeratac:<VER>
```
Comment (maxulysse) [inline modules/nf-core/cellrangeratac/Dockerfile:2]:
```suggestion
# Push to quay.io/nf-core/cellrangeratac:<VER>
```

File: modules/nf-core/cellrangeratac/mkfastq/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,40 @@
+# Dockerfile to create container with Cell Ranger ATAC v2.1.0 and bcl2fastq v2.20.0
+# Push to nfcore/cellranger-atac-mkfastq:<VER>
```
Comment (maxulysse) [inline modules/nf-core/cellrangeratac/mkfastq/Dockerfile:2]:
```suggestion
# Push to quay.io/nf-core/cellranger-atac-mkfastq:<VER>
```

File: modules/nf-core/cellrangeratac/count/main.nf [hunk 4]
```diff
@@ -0,0 +1,54 @@
+process CELLRANGERATAC_COUNT {
+    tag "$meta.id"
+    label 'process_high'
+
+    container "nfcore/cellranger-atac:2.1.0"
```
Comment (maxulysse) [inline modules/nf-core/cellrangeratac/count/main.nf:5]:
```suggestion
    container "nf-core/cellranger-atac:2.1.0"
```

File: modules/nf-core/cellrangeratac/mkfastq/main.nf [hunk 5]
```diff
@@ -0,0 +1,47 @@
+process CELLRANGERATAC_MKFASTQ {
+    tag "mkfastq"
+    label 'process_medium'
+
+    container "nfcore/cellranger-atac:2.1.0"
```
Comment (maxulysse) [inline modules/nf-core/cellrangeratac/mkfastq/main.nf:5]:
```suggestion
    container "nf-core/cellranger-atac:2.1.0"
```

File: modules/nf-core/cellrangeratac/mkref/main.nf [hunk 6]
```diff
@@ -0,0 +1,39 @@
+process CELLRANGERATAC_MKREF {
+    tag "$reference_config"
+    label 'process_medium'
+
+    container "nfcore/cellranger-atac:2.1.0"
```
Comment (maxulysse) [inline modules/nf-core/cellrangeratac/mkref/main.nf:5]:
```suggestion
    container "nf-core/cellranger-atac:2.1.0"
```

File: modules/nf-core/cellrangeratac/README.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+2. Edit the Dockerfile. Update the Cell Ranger versions in this line:
+
+   ```bash
+   ENV CELLRANGERATAC_VER=<VERSION>
+   ```
+
+3. Create and test the container:
+
+   ```bash
+   docker build . -t nfcore/cellranger-atac:<VERSION>
```
Comment (maxulysse) [inline modules/nf-core/cellrangeratac/README.md:19]:
```suggestion
   docker build . -t quay.io/nf-core/cellranger-atac:<VERSION>
```

File: modules/nf-core/cellrangeratac/README.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+   ```
+
+4. Access rights are needed to push the container to the Dockerhub nfcore organization, please ask a core team member
+   to do so.
+
+   ```bash
+   docker push nfcore/cellranger-atac:<VERSION>
```
Comment (maxulysse) [inline modules/nf-core/cellrangeratac/README.md:26]:
```suggestion
   docker push quay.io/nf-core/cellranger-atac:<VERSION>
```

File: modules/nf-core/cellrangeratac/mkfastq/README.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+   ```bash
+   ENV BCL2FASTQ2_VER=<VERSION> \
+       ATAC_VER=<VERSION>
+   ```
+
+3. Create and test the container:
+
+   ```bash
+   docker build . -t nfcore/cellranger-atac-mkfastq:<CELLRANGERATAC_VERSION>
```
Comment (maxulysse) [inline modules/nf-core/cellrangeratac/mkfastq/README.md:19]:
```suggestion
   docker build . -t quay.io/nf-core/cellranger-atac-mkfastq:<CELLRANGERATAC_VERSION>
```

File: modules/nf-core/cellrangeratac/mkfastq/README.md [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+provided for the cellranger nf-core module is not provided nor supported by either Illumina or 10x Genomics. Updating
+the bcl2fastq2 or Cell Ranger ATAC versions in the container and pushing the update to Dockerhub needs to be done manually.
+
+1. Navigate to the appropriate download pages. - [bcl2fastq2](https://emea.support.illumina.com/sequencing/sequencing_software/bcl2fastq-conversion-software.html): download the linux rpm installer of the desired bcl2fastq2 version with `curl` or `wget`. Place this file in the same folder where the Dockerfile lies. - [Cell Ranger ATAC](https://support.10xgenomics.com/single-cell-atac/software/pipelines/latest/installation): download the tar ball of the desired Cell Ranger ATAC version with `curl` or `wget`. Place this file in the same folder where the Dockerfile lies.
+
+2. Edit the Dockerfile. Update the bcl2fastq2 and Cell Ranger versions in this line:
+
+   ```bash
+   ENV BCL2FASTQ2_VER=<VERSION> \
+       ATAC_VER=<VERSION>
```
Comment (maxulysse) [inline modules/nf-core/cellrangeratac/mkfastq/README.md:13]:
```suggestion
       CELLRANGERATAC_VER=<VERSION>
```

**Comment by Emiller88** [comment]
Let's make an issue to rewrite the tests in nf-test(or you can here if you want)

**Comment by maxulysse** [comment]
Solution I approve by @Emiller88 .
we merge as is, and we fix when we port to nf-test

**Pull request merged**